nni/mutable/utils.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+__all__ = [
+    'uid', 'reset_uid',
+    'NoContextError', 'ContextStack', 'get_current_context',
+    'LabelNamespace', 'auto_label',
+]
+
+import logging
+from collections import defaultdict
+from typing import Any
+
+_last_uid = defaultdict(int)
+
+_logger = logging.getLogger(__name__)
+
+
+def uid(namespace: str = 'default') -> int:
+    """Global counter for unique id. Not thread-safe."""
+    _last_uid[namespace] += 1
+    return _last_uid[namespace]
```

Comment:
does it support experiment resume?

Reply:
Experiment resume is not related to this part.